agb/src/display/tiled/vram_manager.rs

```diff
@@ -144,6 +167,46 @@ impl VRamManager {
         TileReference(NonNull::new(ptr as *mut _).unwrap())
     }
 
+    pub fn new_dynamic_tile<'a>(&mut self) -> DynamicTile<'a> {
+        let tile_format = TileFormat::FourBpp;
+        let new_reference: NonNull<u32> =
+            unsafe { TILE_ALLOCATOR.alloc(TILE_LAYOUT) }.unwrap().cast();
+        let tile_reference = TileReference(new_reference);
+
+        let index = Self::index_from_reference(tile_reference);
+
+        let tiles = unsafe {
+            slice::from_raw_parts_mut(TILE_RAM_START as *mut u8, 1024 * tile_format.tile_size())
+        };
+
+        let tile_set = TileSet::new(tiles, tile_format);
+
+        self.tile_set_to_vram.insert(
+            TileInTileSetReference::new(&tile_set, index as u16),
+            tile_reference,
+        );
+
+        self.reference_counts.resize(
+            self.reference_counts.len().max(index + 1),
+            Default::default(),
+        );
+        self.reference_counts[index] =
+            TileReferenceCount::new(TileInTileSetReference::new(&tile_set, index as u16));
+
+        DynamicTile {
+            tile_data: &mut tiles
+                [index * tile_format.tile_size()..(index + 1) * tile_format.tile_size()],
```

Comment:
@corwinkuiper I'm slightly worried about the non-volatile nature of this... any thoughts?